2.train-VAE/betavae_multilayer.py

```diff
@@ -0,0 +1,267 @@
+# Code based on https://github.com/1Konny/Beta-VAE/blob/master/model.py
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import pandas as pd 
+import numpy as np
+import pathlib
+
+
+class BetaVAEMulti(nn.Module):
+    """
+    Beta Variational Autoencoder (VAE) class with multiple layers.
+
+    Args:
+        input_dim (int): Dimension of the input data.
+        latent_dim (int): Dimension of the latent space.
+        beta (float): Weight for the Kullback-Leibler divergence term in the loss function.
+        hidden_dim (int): Dimension of the hidden layer. (utilized for multiple layers)
+        num_layers (int): Number of layers
+    """
+
+    def __init__(self, input_dim, latent_dim, beta, hidden_dim, num_layers):
+        super(BetaVAEMulti, self).__init__()
+        self.latent_dim = latent_dim
+        self.beta = beta
+
+        # Create encoder with variable number of layers
+        layers = []
+        in_dim = input_dim
+        for _ in range(num_layers):
+            layers.append(nn.Linear(in_dim, hidden_dim))
+            layers.append(nn.BatchNorm1d(hidden_dim))
+            layers.append(nn.ReLU())
+            in_dim = hidden_dim
```

Comment:
The hidden layers all have the same amount of nodes. Would it be worth while to change this to a variable amount of nodes??

Reply:
Might be! will try it out